src/background/Wallet/Wallet.ts

```diff
@@ -1047,11 +1038,8 @@ export class Wallet {
     const networksStore = getNetworksStore(Model.getPreferences(this.record));
     const networks = await networksStore.loadNetworksByChainId(chainId);
     const nodeUrl = networks.getRpcUrlInternal(networks.getChainById(chainId));
-    if (FEATURE_PAYMASTER_ENABLED) {
-      return new ZksProvider(nodeUrl);
-    } else {
-      return new ethers.providers.JsonRpcProvider(nodeUrl);
-    }
+    // TODO: should we conditionally use regular provider for non-paymaster txs?
+    return new ZksProvider(nodeUrl); // return new ethers.providers.JsonRpcProvider(nodeUrl);
```

Comment:
We need ensure `ZksProvider` works for all transactions (paymaster/non-paymaster)... I’m not sure about the implementation differences between these two providers, but from the first sight, this doesn’t look entirely safe to me :)